hoomd/hpmc/external/field.py

```diff
@@ -1,243 +1,223 @@
+            self.reference_orientations,
+            self.k_rotational,
+            self.symmetries,
+        )
+        super()._attach()
+
+    @log(requires_run=True)
+    def energy(self):
+        """float: The energy of the harmonic field :math:`[\\mathrm{energy}]`.  # noqa
 
-    def get_energy(self):
-        R"""    Get the current energy of the lattice field.
-                This is a collective call and must be called on all ranks.
-        Example::
-            mc = hpmc.integrate.sphere(seed=415236);
-            lattice = hpmc.field.lattice_field(mc=mc, position=fcc_lattice, k=1000.0);
-            run(20000)
-            eng = lattice.get_energy()
         """
-        timestep = hoomd.context.current.system.getCurrentTimeStep()
-        return self.cpp_compute.getEnergy(timestep)
+        timestep = self._simulation.timestep
+        return self._cpp_obj.getEnergy(timestep, True, True)
 
-    def get_average_energy(self):
-        R"""    Get the average energy per particle of the lattice field.
-                This is a collective call and must be called on all ranks.
-
-        Example::
-            mc = hpmc.integrate.sphere(seed=415236);
-            lattice = hpmc.field.lattice_field(mc=mc, position=fcc_lattice, k=exp(15));
-            run(20000)
-            avg_eng = lattice.get_average_energy() //  should be about 1.5kT
+    @log(requires_run=True)
+    def energy_translational(self):
+        """float: The energy associated with positional fluctuations \
+            :math:`[\\mathrm{energy}]`.  # noqa
 
         """
-        timestep = hoomd.context.current.system.getCurrentTimeStep()
-        return self.cpp_compute.getAvgEnergy(timestep)
+        timestep = self._simulation.timestep
+        return self._cpp_obj.getEnergy(timestep, True, False)
 
-    def get_sigma_energy(self):
-        R"""    Gives the standard deviation of the average energy per particle of the lattice field.
-                This is a collective call and must be called on all ranks.
+    @log(requires_run=True)
+    def energy_rotational(self):
+        """float: The energy associated with rotational fluctuations \
+            :math:`[\\mathrm{energy}]`.
 
-        Example::
-            mc = hpmc.integrate.sphere(seed=415236);
-            lattice = hpmc.field.lattice_field(mc=mc, position=fcc_lattice, k=exp(15));
-            run(20000)
-            sig_eng = lattice.get_sigma_energy()
+        This function will always return 0 for a non-orientable shape.
 
         """
-        timestep = hoomd.context.current.system.getCurrentTimeStep()
-        return self.cpp_compute.getSigma(timestep)
+        timestep = self._simulation.timestep
+        return self._cpp_obj.getEnergy(timestep, False, True)
 
 
-class external_field_composite(_external):
-    R""" Manage multiple external fields.
+class external_field_composite(_External):  # noqa
+    R"""Manage multiple external fields.  # noqa: this will change in v3
```

Comment:
```suggestion
    r"""Manage multiple external fields.
```
Don't include noqa in docstrings.

I notice there are many more docstrings that include noqa, please fix all of them.

Reply:
Most of these are left overs from v2. For those I would recommend removal. Most of these should cause errors. For those that may be useful for v3, we can do as you suggest for walls and for the rest using noqa seems appropriate to not create work that will likely have to be redone.

For the classes and functions ported though, they should not have noqa.

Reply:
A lot of the `noqa`s are indeed in docs for classes that have not been ported over to v3 yet. I removed several of those classes and there are no more `noqa`s in the classes relevant to this PR.